### ISSUE TITLE
[Bug] Downloads Stop/Crashes on Downloading Large List from Artist on Window 11 24H2 using GUI All-In-One App

### ISSUE THREAD

**Issue by HanSolo71** [exislow/tidal-dl-ng#277]
### What happened?

Example artist: Florence + The Machine. After a large number of albums and singles are downloaded via "Download Artist" the download process stops at "Downloading 0%" after some number of songs. 

### Version App

v0.21.4

### What operating system are you seeing the problem on?

Windows

### Your settings

```json
{
    "skip_existing": true,
    "lyrics_embed": false,
    "lyrics_file": false,
    "video_download": false,
    "download_delay": true,
    "download_base_path": "G:\\Music",
    "quality_audio": "HIGH",
    "quality_video": "1080",
    "format_album": "Albums/{album_artist}/{album_artist} - {album_title}{album_explicit}/{track_volume_num_optional}{album_track_num}. {artist_name} - {track_title}{album_explicit}",
    "format_playlist": "Playlists/{playlist_name}/{artist_name} - {track_title}",
    "format_mix": "Mix/{mix_name}/{artist_name} - {track_title}",
    "format_track": "Tracks/{album_artist}/{album_artist} - {album_title}{album_explicit}/{track_volume_num_optional}{album_track_num}. {artist_name} - {track_title}{album_explicit}",
    "format_video": "Videos/{artist_name} - {track_title}{track_explicit}",
    "video_convert_mp4": false,
    "path_binary_ffmpeg": "C:\\Program Files\\ffmpeg\\bin\\ffmpeg.exe",
    "metadata_cover_dimension": "1280",
    "metadata_cover_embed": true,
    "cover_album_file": true,
    "extract_flac": true,
    "downloads_simultaneous_per_track_max": 20,
    "download_delay_sec_min": 3.0,
    "download_delay_sec_max": 5.0
}
```

**Comment by exislow** [comment]
I cannot help you if you do not a) provide any proof, b) logs and c) any very specific information how to re-produce this.

**Comment by HanSolo71** [comment]
The all in one app does not appear to create logs, I tried using procmon to find a log file being written and it appears it is not written. That is why i did not give logs.

As for reproducing. Select > Artist > Florence + the Machine > Download and wait.

**Comment by exislow** [comment]
I do not know, if you mean by "the all in one app". Logs are either written on the console (if you use the pip installation) or in the middle lower "text window" of the GUI. You can copy from both. Please also provide the exact number of songs, when it crashes. And if you are using the GUI, please also provide a screenshot.

**Comment by HanSolo71** [comment]
There are no errors. Next time it happens I will grab you the output. It simply stalls at 0%.

**Comment by HanSolo71** [comment]
While I am this, do you want a process dump when the app hangs also?

**Comment by exislow** [comment]
It sometimes hangs, while closing the app, right? I do not need that process dump. It has something to do with pyInstaller. Need to investigate that framework sometime.

**Comment by HanSolo71** [comment]
No, application stays usable, IE, I can add new items to download, but downloading stops and hangs.

**Comment by exislow** [comment]
I think there is some sort of race condition or infinity loop. I doubt something has crashed in this case. But if you can get a process dum easily, sure why not :-)

**Comment by geneware** [comment]
Similar problem.  I try to download "The Osmonds - 20th Century Masters...". GUI 0.24.6 says download started, and nothing happens.
I x out of the GUI, and in Windows process manager I see the download running at 100%, totally hung up.
But... when I open the album in Tidal, it is empty.  No tracks.
Need a fix for this.  Also would be nice to have a cancel download button, or for the "clear all" button to clear active downloads.

**Comment by exislow** [comment]
If you like to share information for re-production of your errors, please share the URLs of the items (right click -> copy url).

**Comment by geneware** [comment]
Copy URL is saying "Failed to set mime data on clipboard".  So can't copy.
But do an album search for "The Osmonds - 20th Century Masters" and 3 are found.
Try downloading all of them, you should see it hang on the empty album.  Look in Windows Task Manager, and the Python process is  running at max CPU, going nowhere.
I reproduced this today.  Maybe just need a check for empties, and skip over them?

**Comment by exislow** [comment]
@geneware: Thank you for this analysis. This will help a lot to pin point the bug. 

Actually I cannot find any results if I look for "The Osmonds - 20th Century Masters". But I have fixed the "Copy URL" bug. Could you please get the latest version (min. v0.25.0) of this app and provide the "Copy URL" output of the empty album?

**Comment by geneware** [comment]
Sorry, we are going down a rabbit hole.
I got the 0.25.0 source code, and just like I did with 0.24.6, I installed
with the suggested commands:
poetry lock
poetry install --all-extras --with dev,docs
All seemed to work.  And now run the program by
poetry run tidal-dl-ng-gui

But I get the config error below.   I then point to the 0.24.6 dir, and it
runs as usual. Config file names in both dirs appear the same. Help!

D:\Downloads\tidal0250-dl-ng>poetry run tidal-dl-ng-gui
Traceback (most recent call last):
  File "D:\Downloads\tidal0250-dl-ng\.venv\Scripts\\tidal-dl-ng-gui", line
3, in <module>
    from tidal_dl_ng.gui import gui_activate
  File "D:\Downloads\tidal0250-dl-ng\tidal_dl_ng\gui.py", line 52, in
<module>
    from config import HandlingApp
ModuleNotFoundError: No module named 'config'

**Comment by exislow** [comment]
I just released a fix for this. Thanks for reporting.

**Comment by geneware** [comment]
Progress, 0.25.3 worked for me.  Here is the URL of the empty album
https://tidal.com/browse/album/36295296
If I try to download it, nothing happens but Task Manager shows a Python
process running 100% cpu on one core.
I hit your new pause button, no change.  Have to close the gui and the cmd
windows to stop the Python process.

I have downloaded thousands of albums, including lists of several hundred
to download with one click.  Only 2 failures, due to empty titles.
Thanks for a great program!

**Comment by exislow** [comment]
This album is not empty. It has one track in it. But it is not available in my country. Maybe in yours, too.

<img width="1375" alt="Image" src="https://github.com/user-attachments/assets/5bc94533-5351-4d87-8cf0-59cedea97f2f" />

I added a check for that. Hopefully it resolves your issue. I cannot really test it, since I cannot download this album due to regional restrictions by TIDAL.
